usaspending_api/agency/v2/views/federal_account_list.py

```diff
@@ -58,20 +68,42 @@ def format_results(self, rows):
     def get(self, request: Request, *args: Any, **kwargs: Any) -> Response:
         self.sortable_columns = ["name", "obligated_amount", "gross_outlay_amount"]
         self.default_sort_column = "obligated_amount"
-        results = self.format_results(self.get_federal_account_list())
-        page_metadata = get_pagination_metadata(len(results), self.pagination.limit, self.pagination.page)
+        results = self.format_results(self.get_tbr_from_file_a_queryset(), self.get_federal_account_list_from_file_b())
+        page_metadata = get_pagination_metadata(len(results), len(results), self.pagination.page)
+        print(page_metadata)  # debug
         return Response(
             {
                 "toptier_code": self.toptier_code,
                 "fiscal_year": self.fiscal_year,
                 "page_metadata": page_metadata,
-                "results": results[self.pagination.lower_limit : self.pagination.upper_limit],
+                "results": results,
                 "messages": self.standard_response_messages,
             }
         )
 
-    def get_federal_account_list(self) -> List[dict]:
+    def get_tbr_from_file_a_queryset(self):
+        """
+        Query Total Budgetary Resources from File A
+        """
+        filters, bureau_info_subquery = self.get_common_query_objects("treasury_account_identifier")
+
+        results = (
+            (AppropriationAccountBalances.objects.filter(*filters))
```

Comment:
Will also need to filter by the `is_final_balalance_of_fy` to make sure that you are not getting all of the TBR values associated.

Or better yet, follow the example used in the `get_federal_account_list_from_file_b` below where the latest submissions_ids are retrieved and then used to determine which records to view.

The problem here is that the `_cpe` value stands for Current Period Ending. Since this value is a sum up to that point it will vary period to period for an agency across a single Fiscal Year.

Reply:
I added a filter on the `submission_ids` that are returned in the File A results in commit d145e39d6de95e5714301d53ccb7f0693773da12. Also, I moved the `get_latest_submission_ids_for_fiscal_year()` function call so that it's only run once per request.